plugins/all_test.go

```diff
@@ -0,0 +1,112 @@
+package all
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// exceptionMap holds those plugins which differ in conventions when defining plugins.
+// Most plugins follow the convention of pkg-name to plugin-name.
+// For ex, Pivot processor pkg github.com/influxdata/telegraf/plugins/processors/pivot maps directly to
+// the last element of the pkg i.e "pivot"
+// But in case of "aws_ec2" processor, the pkg is defined as "github.com/influxdata/telegraf/plugins/processors/aws/ec2".
+// This ensure package names are not tied with plugin names.
+// it should be of the form <pkg-name>: <plugin-name>
+var exceptionMap = map[string]string{
+	"github.com/influxdata/telegraf/plugins/processors/aws/ec2": "aws_ec2",
+}
+
+func Test_AllPlugins(t *testing.T) {
+	pluginDirs := []string{"aggregators", "inputs", "outputs", "parsers", "processors", "secretstores"}
+	for _, dir := range pluginDirs {
+		testPluginDirectory(t, dir)
+	}
+}
+
+func testPluginDirectory(t *testing.T, directory string) {
+	allDir := filepath.Join(directory, "all")
+	err := filepath.WalkDir(allDir, func(goPluginFile string, d fs.DirEntry, walkErr error) error {
+		require.NoError(t, walkErr)
+		if d.IsDir() || strings.HasSuffix(d.Name(), "_test.go") {
+			return nil
+		}
+		parseSourceFile(t, goPluginFile, directory)
```

Comment:
What do you think of moving the Run to here?
```suggestion
		t.Run(goPluginFile, func(t *testing.T) {
			parseSourceFile(t, goPluginFile, directory)
		}
```

Then it will become easier to add other tests for this file in the future.